src/test/java/org/broadinstitute/hellbender/tools/spark/sv/CollectISizeDistributionSparkTest.java

```diff
@@ -0,0 +1,187 @@
+package org.broadinstitute.hellbender.tools.spark.sv;
+
+import htsjdk.samtools.metrics.MetricsFile;
+import org.broadinstitute.hellbender.CommandLineProgramTest;
+import org.broadinstitute.hellbender.utils.test.ArgumentsBuilder;
+import org.broadinstitute.hellbender.utils.test.BaseTest;
+import org.broadinstitute.hellbender.tools.picard.analysis.InsertSizeMetrics;
+import org.testng.Assert;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+
+import java.io.File;
+import java.io.FileReader;
+import java.io.IOException;
+
+public final class CollectISizeDistributionSparkTest extends CommandLineProgramTest{
+
+    private static final File TEST_DATA_DIR = new File(getTestDataDir(), "picard/analysis/CollectInsertSizeMetrics");
+
+    public String getTestedClassName() {
+        return CollectISizeDistributionSpark.class.getSimpleName();
+    }
+
+    @DataProvider(name="metricsfiles")
+    public Object[][] insertSizeMetricsFiles() {
+        return new Object[][] {
+                {"insert_size_metrics_test.sam", null},
+                {"insert_size_metrics_test.bam", null},
+                {"insert_size_metrics_test.cram", hg19_chr1_1M_Reference}
+        };
+    }
+
+    @Test(dataProvider="metricsfiles")
+    public void test(final String fileName, final String referenceName) throws IOException {
+        final File input = new File(TEST_DATA_DIR, fileName);
+        final File outfile = BaseTest.createTempFile("test", ".insert_size_metrics");
+        final File pdf = BaseTest.createTempFile("test", ".pdf");
+
+        final ArgumentsBuilder args = new ArgumentsBuilder();
+        args.add("--input");
+        args.add(input.getAbsolutePath());
+        args.add("--output");
+        args.add(outfile.getAbsolutePath());
+        if (null != referenceName) {
+            final File REF = new File(referenceName);
+            args.add("-R");
+            args.add(REF.getAbsolutePath());
+        }
+        args.add("--HISTOGRAM_FILE");
+        args.add(pdf.getAbsolutePath());
+
+        runCommandLine(args.getArgsArray());
+
+        final MetricsFile<InsertSizeMetrics, Comparable<?>> output = new MetricsFile<>();
+        output.read(new FileReader(outfile));
+
+        for (final InsertSizeMetrics metrics : output.getMetrics()) {
```

Comment:
as we both saw, this test code is incorrect because will not detect that the insert_size_metrics_test.bam is empty . Please change it so that the actual results are compared to a pre-computed file (the same file for sam/bam/cram)